File: app/src/main/java/com/task/paginationlist/PaginationListApplication.java
package com.task.paginationlist;

import android.app.Application;

import com.task.paginationlist.di.components.AppComponent;
import com.task.paginationlist.di.components.DaggerAppComponent;
import com.task.paginationlist.di.modules.RepositoryModule;

public class PaginationListApplication extends Application {

    private static AppComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        component = buildAppComponent();
    }

    private AppComponent buildAppComponent() {
        return DaggerAppComponent.builder().repositoryModule(new RepositoryModule(this)).build();
    }

    public static AppComponent getComponent() {
        return component;
    }
}

File: app/src/main/java/com/task/paginationlist/PaginationListApplication.kt
package com.task.paginationlist

import android.app.Application
import com.task.paginationlist.di.components.AppComponent
import com.task.paginationlist.di.components.DaggerAppComponent
import com.task.paginationlist.di.modules.RepositoryModule

class PaginationListApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        component = buildAppComponent()

    }

    private fun buildAppComponent(): AppComponent {
        return DaggerAppComponent.builder().repositoryModule(RepositoryModule(this)).build()
    }

    companion object {

        var component: AppComponent? = null
            private set
    }
}
